hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -118,7 +118,7 @@ object HoodieSparkSqlWriter {
     } else {
       // Handle various save modes
       handleSaveModes(sqlContext.sparkSession, mode, basePath, tableConfig, tblName, operation, fs)
-      val partitionColumns = HoodieWriterUtils.getPartitionColumns(keyGenerator)
+      val partitionColumns = HoodieSparkWriterUtils.getPartitionColumnsFromKeyGenerator(keyGenerator, toProperties(parameters))
```

Comment:
why change the name? you can just keep `getPartitionColumns()` given teh args are different anyway? encoding the arg type in the method name is a no-no in my book :)

Reply:
Fixed.